section1/reflection.md

```diff
@@ -1,19 +1,91 @@
+    descriptive names. Variable names can contain letters, numbers, underscores,
+    and dollar signs. Variable names can begin with letters, underscores, or dollar signs.
+    Variable names are CASE-sensitive. (a and A are different variables) No spaces are
+    left between variable names. JavaScript keywords cannot be used as variable names.
+    (var, debugger, if, while…etc.) Let's say we have a box where we put some Lego pieces.
+    And we want to make a new lego using the piece we want. Maybe a spider-headed
+    person, perhaps a woman with flower arms, a man with a dog's paw. Let's say
+    variable to the box where these legos pieces are, and the variable name is
+    each lego piece chosen by the child; value is what he will use that piece as.
+    Thus, it will be shown that lego's box is our variable where we store lego's pieces.
+    And the child, whenever they want, pick any piece and give name/task.
+    
+
+    6. Think of a site or app you use frequently. What are three variables that are probably
+     used? Which data type would each of those variables probably hold?
+
+    - _I can give examples of many different applications amazon, chewy, Instagram, Petco.
+    While taking an appointment from a groomer in Petco, possible used variables are name, age,
+    gender, and neutered information about my pet. My account, password, and user name are necessary.
+    When they filled the name and title while getting a groomer appointment, this was an example of
+    a string-type data type. An example is the number data type when they ask for age. If the dog is
+    infertile, I can give an example of the boolean data type with true and false answers._
+
+    7. In your own words, explain what concatenation is.
+
+    - **The concatenation** allows us to create a single value by joining two or more strings together.
+    "Concatenate" is a programming term that String data type that is logged to console by adding new words and using variable value.
+
+    8. Think of a site or app you use frequently. Where do you think the developers used concatenation?
+
+    - I've been thinking about an example when taking a covid test these days. After answering the necessary
+     information in the question's name, the last name such as string data type, age such as number data type,
+     and answering for true/false questions such as boolean data type,  I get a confirmation sentence with my
+     information. I think the concatenation is used here.
+
+    9. What questions do you still have about the work we've done so far? (not having a question is not an option)
+
+    - My question is concatenation and interpolation. Maybe I missed the difference between them. While studying,
+     the concatenation(join together) looked the same programming as interpolation. If I missed something between
+     them, I would like to know. **Same result, different way!**
```

Comment:
`Same result, different way` is a good way to think of it.

```js
var speedy = "quick red fox";
var slowPoke = "lazy brown dog";

console.log(`The ${speedy} jumped over the ${slowPoke}.`);
console.log("The " + speedy + " jumped over the " + slowPoke + ".");
```

Convention in the JS community generally favors string interpolation, because it's just like reading/writing a sentence.

Concatenation can be harder to read as code because our brains have to parse multiple strings and piece them together into one sentence.

As you get further into the program, you'll see that sometimes we _have_ to use concatenation for technical reasons (but don't worry too much about this now). So it's important to know and understand both ways!